Tests/src/PublicAPITests/ComponentTests/CinematicCameraComponentTests.cpp

```diff
@@ -0,0 +1,237 @@
+
+	auto& SystemsManager = csp::systems::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* Connection	 = SystemsManager.GetMultiplayerConnection();
+	auto* EntitySystem	 = SystemsManager.GetSpaceEntitySystem();
+
+	const char* TestSpaceName		 = "OLY-UNITTEST-SPACE-REWIND";
+	const char* TestSpaceDescription = "OLY-UNITTEST-SPACEDESC-REWIND";
+
+	char UniqueSpaceName[256];
+	SPRINTF(UniqueSpaceName, "%s-%s", TestSpaceName, GetUniqueString().c_str());
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space
+	csp::systems::Space Space;
+	CreateSpace(SpaceSystem, UniqueSpaceName, TestSpaceDescription, csp::systems::SpaceAttributes::Private, nullptr, nullptr, nullptr, Space);
+
+	auto [EnterResult] = AWAIT_PRE(SpaceSystem, EnterSpace, RequestPredicate, Space.Id);
+
+	EXPECT_EQ(EnterResult.GetResultCode(), csp::systems::EResultCode::Success);
+
+	EntitySystem->SetEntityCreatedCallback(
+		[](csp::multiplayer::SpaceEntity* Entity)
+		{
+		});
+
+	// Create object to represent the Camera
+	csp::common::String ObjectName = "Object 1";
+	SpaceTransform ObjectTransform = {csp::common::Vector3::Zero(), csp::common::Vector4::Zero(), csp::common::Vector3::One()};
+	auto [CreatedObject]		   = AWAIT(EntitySystem, CreateObject, ObjectName, ObjectTransform);
+
+	// Create Camera component
+	auto* CinematicCamera = static_cast<CinematicCameraSpaceComponent*>(CreatedObject->AddComponent(ComponentType::CinematicCamera));
+
+	// Ensure defaults are set
+	EXPECT_FLOAT_EQ(CinematicCamera->GetFocalLength(), 0.035f);
+	EXPECT_EQ(CinematicCamera->GetPosition(), csp::common::Vector3::Zero());
+	EXPECT_EQ(CinematicCamera->GetRotation(), csp::common::Vector4(0, 0, 0, 1));
+	EXPECT_EQ(CinematicCamera->GetScale(), csp::common::Vector3::One());
+	EXPECT_FLOAT_EQ(CinematicCamera->GetAspectRatio(), 1.778f);
+	EXPECT_EQ(CinematicCamera->GetSensorSize(), csp::common::Vector2(0.036f, 0.024f));
+	EXPECT_FLOAT_EQ(CinematicCamera->GetNearClip(), 0.1f);
+	EXPECT_FLOAT_EQ(CinematicCamera->GetFarClip(), 20000.0f);
+	EXPECT_FLOAT_EQ(CinematicCamera->GetIso(), 400.0f);
+	EXPECT_FLOAT_EQ(CinematicCamera->GetShutterSpeed(), 0.0167);
+	EXPECT_FLOAT_EQ(CinematicCamera->GetAperture(), 4.0f);
+	EXPECT_FALSE(CinematicCamera->GetIsViewerCamera());
+
+
+	CinematicCamera->SetFocalLength(2.0f);
+	CinematicCamera->SetPosition(csp::common::Vector3(3, 2, 1));
+	CinematicCamera->SetRotation(csp::common::Vector4(1, 2, 3, 1));
+	// This value should be ignored
+	CinematicCamera->SetScale(csp::common::Vector3(1, 2, 3));
+	CinematicCamera->SetAspectRatio(1.3f);
+	CinematicCamera->SetSensorSize(csp::common::Vector2(1, 2));
+	CinematicCamera->SetNearClip(1.0f);
+	CinematicCamera->SetFarClip(100.0f);
+	CinematicCamera->SetIso(1000.0f);
+	CinematicCamera->SetShutterSpeed(0.003f);
+	CinematicCamera->SetAperture(10.0f);
+	CinematicCamera->SetIsViewerCamera(true);
+
+
+	// Set new values
```

Comment:
This comment belongs above when actually setting the values. We are checking they are as expected here